guvnor-services/guvnor-services-backend/src/main/java/org/guvnor/common/services/backend/file/upload/AbstractFileServlet.java

```diff
@@ -294,4 +297,12 @@ protected void processAttachmentDownload( final String url,
         }
     }
 
+    protected String getSessionId( final HttpServletRequest request,
+                                   final SessionProvider sessionProvider ) {
+        final HttpSession session = request.getSession( true );
+        final String clientId = request.getParameter( "clientId" );
```

Comment:
I think there is (at least) another "file upload" screen (in ```kie-wb-common```) that you have not (yet) fixed to include a ```clientId```. Therefore there is potential for this to be ```null``` which may cause problems with ```SessionProvider``` (probably will!). I'll try to find the screen - but for now please check for *all* usage of ```AttachmentFileWidget```